File: webui/src/main/scala/com/taskadapter/webui/LogFinder.scala
package com.taskadapter.webui

import java.io.File

import org.apache.log4j.RollingFileAppender

object LogFinder {
  def getLogFileLocation: String = {
    val logger = org.apache.log4j.Logger.getRootLogger
    val allAppenders = logger.getAllAppenders
    while (allAppenders.hasMoreElements) {
      val e = allAppenders.nextElement
      if (e.isInstanceOf[RollingFileAppender]) {
        // found it
        val filePath = e.asInstanceOf[RollingFileAppender].getFile
        return new File(filePath).getAbsolutePath
      }
    }
    ""
  }
}

File: webui/src/main/java/com/taskadapter/webui/LogFinder.java
package com.taskadapter.webui;

import ch.qos.logback.core.rolling.RollingFileAppender;

import java.io.File;

public class LogFinder {
    public static String getLogFileLocation() {
        var logger = org.apache.log4j.Logger.getRootLogger();
        var allAppenders = logger.getAllAppenders();
        while (allAppenders.hasMoreElements()) {
            var e = allAppenders.nextElement();
            if (e instanceof RollingFileAppender) {
                // found it
                var filePath = ((RollingFileAppender) e).getFile();
                return new File(filePath).getAbsolutePath();
            }
        }
        return "";
    }
}
